content/en/llm_observability/setup/sdk/nodejs.md

```diff
@@ -0,0 +1,713 @@
+{{< /code-block >}}
+
+
+### Submit evaluations
+
+`llmobs.submitEvaluation()` can be used to submit your custom evaluation associated with a given span.
+
+#### Arguments
+
+The `llmobs.submitEvaluation()` method accepts the following arguments:
+
+`span_context`
+: required - _dictionary_
+<br />The span context to associate the evaluation with. This should be the output of `LLMObs.export_span()`.
+
+`evaluationOptions`
+: required - _object_
+<br />An object of the evaluation data.
+
+The `evaluationOptions` object can contain the following:
+
+`label`
+: required - _string_
+<br />The name of the evaluation.
+
+`metricType`
+: required - _string_
+<br />The type of the evaluation. Must be one of "categorical" or "score".
+
+`value`
+: required - _string or numeric type_
+<br />The value of the evaluation. Must be a string (for categorical `metric_type`) or number (for score `metric_type`).
+
+`tags`
+: optional - _dictionary_
+<br />A dictionary of string key-value pairs that users can add as tags regarding the evaluation. For more information about tags, see [Getting Started with Tags][9].
+
+### Example
+
+{{< code-block lang="javascript" >}}
+function llmCall () {
+  const completion = ... // user application logic to invoke LLM
+  const spanContext = llmobs.exportSpan()
+  llmobs.submitEvaluation(spanContext, {
+    label: "harmfulness",
+    metricType: "score",
+    value: 10,
+    tags: { evaluationProvider: "ragas" }
+  })
+  return completion
+}
+llmCall = llmobs.wrap('llm', { name: 'invokeLLM', modelName: 'claude', modelProvider: 'anthropic' }, llmCall)
+{{< /code-block >}}
+
+## Advanced tracing
+
+### Tracing spans using inline methods
+
+The `llmobs` SDK provides a corresponding inline method to automatically trace the operation a given code block entails. These methods have the same argument signature as their function wrapper counterparts, with the addition that `name` defaults to the span kind (`llm`, `workflow`, and so on) if not provided. This method will finish the span under the following conditions:
+
+1. The function returns a Promise, in which case the span will finish when the promise is resolved or rejected.
+2. The function takes a callback as its last parameter, in which case the span will finish when that callback is called.
+3. The function doesn't accept a callback and doesn't return a Promise, in which case the span will finish at the end of the function execution.
+
+Additionally, the return value of this function is the return value of the function provided.
+
+#### Example
+
+{{< code-block lang="javascript" >}}
+function processMessage () {
+  return llmobs.trace('workflow', { name: 'processMessage', sessionId: '<SESSION_ID>', mlApp: '<ML_APP>' }, workflowSpan => {
+    ... // user application logic
+    return
+  })
+}
+{{< /code-block >}}
+
+### Persisting a span across contexts
+
+To manually start and stop a span across different contexts or scopes:
```

Comment:
Might be worth introducing `llmobs.startSpan()` as a tool provided by the SDK to manually start and stop a span here.

Reply:
could you clarify what you mean by this? I have an example below using `llmobs.startSpan`, is that what you're referring to? for reference, i just followed this section: https://docs.datadoghq.com/llm_observability/setup/sdk/#persisting-a-span-across-contexts

Reply:
I'm just pointing out that users might find it confusing when the first mention of `llmobs.startSpan()` is in step 1 of the below example without any description of what it is/does (I know, should be obvious but it always helps to be 100% clear in these cases). Maybe something along the lines of:
```
The LLM Observability Node.js SDK provides the `llmobs.startSpan()` method to manually start and stop a span to be persisted across different contexts or scopes:
```

Reply:
ahh ok yes that makes sense, will add something like that!